packages/extension/src/ui/apis/asset/permitERC20Allowance.js

```diff
@@ -0,0 +1,54 @@
+import ConnectionService from '~/ui/services/ConnectionService';
+import Web3Service from '~/helpers/Web3Service';
+
+export default async function permitERC20({
+    asset,
+    requestedAllowance,
+    allowanceSpender,
+}) {
+
+
+    const {
+        linkedTokenAddress,
+    } = asset;
+
+    let error;
+    try {
+       const nonce =  await Web3Service
+            .useContract('ERC20')
+            .at(linkedTokenAddress)
+            .method('nonces')
+            .send(
+                allowanceSpender,
+            );
+    } catch (e) {
+        error = e;
+    }
+    const exipry = Date.now() + 24 * 60 * 60;
```

Comment:
```suggestion
    const expiry = Date.now() + 24 * 60 * 60;
```

(this is going to go anyways)